### ISSUE TITLE
Deferred git state: wrong type argument

### ISSUE THREAD

**Issue by hermann-p** [Alexander-Miller/treemacs#312]
Although treemacs is fantastic to work with and imho incredibly superior to the speedbar, it has one negative impact on my workflow: The minibuffer gets spammed with

```
Error running timer ‘treemacs--apply-deferred-git-state’: (wrong-type-argument listp fatal:)
```

messages, so it's virtually unusable.

Setup:
- emacs 26.1-2
- Prelude

Relevant active modes, all latest (2018-12-05) from melpa:
- treemacs-projectile (just started using treemacs, so can't tell if this is new or regression)
- projectile
- tide in typescript-mode
- magit

I have a feeling, these error messages start appearing only after I launch magit; at least their appearance increases greatly once I visit a magit-buffer. As expected they go away, when I disable deferred git state checking.

If you need more information to debug (or it is just a stupid setting error), please let me know.

**Comment by Alexander-Miller** [comment]
> (wrong-type-argument listp fatal:)

Pretty sure I know what's going on. This is an unintended consequence of the recent fix to make treemacs-as-a-side-window work in emacs 26. Now  treemacs sticks around when magit is opened and follow-mode is triggered when magit moves into files in the .git directory. You'll get an error message when you launch git commands inside .git and treemacs fails to parse that. 

Quite a trail, but I am sure I can quickly fix this, I'm probably just missing a status code check somewhere. In the meantime you can just disable `treemacs-display-in-side-window`.

In the near future I also plan to introduce a means to make treemacs go away when a certain command (like magit or dired) is launched even if it is a side-window.

**Comment by hermann-p** [comment]
The workaround disabling `treemacs-display-in-side-window` seems to do the trick for now, thanks for that. 

Error messages disappear but treemacs stays visible just where it used to be with that option enabled. I think I'll go read the docs to understand that variable ;)

**Comment by Alexander-Miller** [comment]
> but treemacs stays visible just where it used to be with that option enabled

Did you kill the treemacs window in the meantime? It only affects the way a new treemacs window is spawned. 

The variable only controls whether emacs will see the treemacs window as a [side window](https://www.gnu.org/software/emacs/manual/html_node/elisp/Side-Windows.html) and side windows survive commands that would usually delete all other windows, like `magit-status` or `delete-other-windows`.

**Comment by Alexander-Miller** [comment]
Turns out I did correctly check the exit code, it's just that the python script that would parse and format git's output would always finish successfully. That's corrected now.
